src/pages/home/index.js

```diff
@@ -0,0 +1,272 @@
+    };
+  }
+
+  renderComponents() {
+    Object.keys(this.components).forEach((component) => {
+      const root = this.subElements[component];
+      const { element } = this.components[component];
+
+      if (element) {
+        root.append(element);
+      }
+    });
+  }
+
+  initEventListeners() {
+    document.addEventListener(
+      "added-to-cart",
+      () => {
+        const productsCount = this.productStore.getProductsCount();
+        const { cartCounter } = this.subElements;
+
+        cartCounter.classList.remove("hidden");
+        cartCounter.innerText = productsCount;
+      },
+      { signal: this.abortController.signal },
+    );
+
+    document.addEventListener(
+      "removed-from-cart",
+      () => {
+        const { cartCounter } = this.subElements;
+        const productsCount = this.productStore.getProductsCount();
+
+        if (productsCount === 0) {
+          cartCounter.classList.add("hidden");
+          cartCounter.innerText = 0;
+        } else {
+          cartCounter.innerText = productsCount;
+        }
+      },
+      { signal: this.abortController.signal },
+    );
+
+    this.components.search.element.addEventListener(
+      "search-filter",
+      (event) => {
+        this.filters.set("_page", "1");
+        this.filters.set("q", event.detail);
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener(
+      "range-selected",
+      (event) => {
+        const { filterName, value } = event.detail;
+
+        this.filters.set("_page", "1");
+
+        const gte = `${filterName}_gte`;
+        const lte = `${filterName}_lte`;
+
+        this.filters.set(gte, value.from);
+        this.filters.set(lte, value.to);
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener("add-filter", (event) => {
+      this.filters.set("_page", "1");
+
+      for (const [key, prop] of new URLSearchParams(event.detail)) {
+        this.filters.append(key, prop);
+      }
+
+      this.updateProductsList();
+    });
+
+    this.components.sideBar.element.addEventListener(
+      "remove-filter",
+      (event) => {
+        const [key, prop] = event.detail.split("=");
+        const filters = this.filters
+          .getAll(key)
+          .filter((item) => item !== prop);
+
+        this.filters.set("_page", "1");
+        this.filters.delete(key);
+
+        for (const filter of filters) {
+          this.filters.append(key, filter);
+        }
+
+        this.updateProductsList();
+      },
+    );
+
+    this.components.sideBar.element.addEventListener("clear-filters", () => {
+      this.resetFilters();
+      this.components.search.clear();
+
+      this.updateProductsList();
+    });
+
+    this.components.pagination.element.addEventListener(
+      "page-changed",
+      (event) => {
+        this.filters.set("_page", event.detail + 1);
+
+        this.updateProductsList();
+      },
+    );
+  }
```

Comment:
**Optimization Issue**: The initEventListeners method adds multiple listeners to the document and components but does not use a consistent mechanism for managing these listeners, which can lead to issues when trying to remove them or when multiple instances of the page component are used. <br> **Fix**: Refactor the event listener management to use a centralized method for adding and removing listeners, which will simplify listener management and ensure that all listeners can be properly cleaned up. <br> **Code Suggestion**: 
 ```
 +    this.eventHandlers = {
 +      addToCart: this.handleAddToCart.bind(this),
 +      removeFromCart: this.handleRemoveFromCart.bind(this),
 +      searchFilter: this.handleSearchFilter.bind(this),
 +      rangeSelected: this.handleRangeSelected.bind(this),
 +      addFilter: this.handleAddFilter.bind(this),
 +      removeFilter: this.handleRemoveFilter.bind(this),
 +      clearFilters: this.handleClearFilters.bind(this),
 +      pageChanged: this.handlePageChanged.bind(this),
 +    };
 +    document.addEventListener("added-to-cart", this.eventHandlers.addToCart);
 +    document.addEventListener("removed-from-cart", this.eventHandlers.removeFromCart);
 +    this.components.search.element.addEventListener("search-filter", this.eventHandlers.searchFilter);
 +    this.components.sideBar.element.addEventListener("range-selected", this.eventHandlers.rangeSelected);
 +    this.components.sideBar.element.addEventListener("add-filter", this.eventHandlers.addFilter);
 +    this.components.sideBar.element.addEventListener("remove-filter", this.eventHandlers.removeFilter);
 +    this.components.sideBar.element.addEventListener("clear-filters", this.eventHandlers.clearFilters);
 +    this.components.pagination.element.addEventListener("page-changed", this.eventHandlers.pageChanged);
 ```